spec/components/guardian_spec.rb

```diff
@@ -1998,16 +1999,29 @@
   end
 
   describe 'can_wiki?' do
+    let(:post) { build(:post) }
+
     it 'returns false for regular user' do
-      expect(Guardian.new(coding_horror).can_wiki?).to be_falsey
+      expect(Guardian.new(coding_horror).can_wiki?(post)).to be_falsey
+    end
+
+    it "returns false when user satisfies trust level but tries to wiki someone else's post" do
+      SiteSetting.min_trust_to_allow_self_wiki = 2
+      expect(Guardian.new(trust_level_2).can_wiki?(post)).to be_falsey
+    end
+
+    it 'returns true when user satisfies trust level and owns the post' do
+      SiteSetting.min_trust_to_allow_self_wiki = 2
+      own_post = Fabricate(:post, user: trust_level_2)
+      expect(Guardian.new(trust_level_2).can_wiki?(own_post)).to be_truthy
     end
```

Comment:
I think we need a test case for `when user does not have the sufficient trust level to wiki his own post`